main/params.go

```diff
@@ -333,14 +334,24 @@ func setNodeConfig(v *viper.Viper) error {
 			dbDir = defaultDbDir
 		}
 		dbDir = os.ExpandEnv(dbDir) // parse any env variables
-		dbPath := path.Join(dbDir, constants.NetworkName(Config.NetworkID), dbVersion)
-		db, err := leveldb.New(dbPath, 0, 0, 0)
+		dbPath := path.Join(dbDir, constants.NetworkName(Config.NetworkID))
+
+		dbManager, err := manager.New(dbPath, dbVersion)
 		if err != nil {
-			return fmt.Errorf("couldn't create db at %s: %w", dbPath, err)
+			return fmt.Errorf("couldn't create db manager at %s: %w", dbPath, err)
 		}
-		Config.DB = db
+		Config.DB = dbManager
 	} else {
-		Config.DB = memdb.New()
+		dbManager, err := manager.NewManagerFromDBs([]*manager.VersionedDatabase{
```

Comment:
We can just do `config.DB, err = manager.NewManagerFromDBs([]*manager.VersionedDatabase{`